### PR TITLE
Getting Started IDE

### PR DESCRIPTION
**Pull request by tobka777**
#728

### CODE REVIEW

**Comment by tobka777** [comment]
My ideas:
1. Installation
     - Installation of devonfw (Link)

2. Constribution
     - Link to Constribution + help for fork
     - Ideas for development e.g. two folders (1. fork, 2. installation) and copy or edit scripts

3. Create new commandlet
Short instructions on how to create a new commandlet with mirrors, autocompletion, doInstall explanation.

4. settings
Setting configuration using ide-settings

5. Tipps and tricks
Description of helpful files such as functions, devon.properties

**Comment by cinnamon-coder-hub** [comment]
Here are some important or helpfull links. 
-----------------------------------------------------------------------------------------------------------------------------------------
@CREITZ25 @tobka777 @alfeilex Please feel free to edit this comment by addind links or linkrelated text, so that we have all links for the file as an overview here. If you are unsure about adding a certain link, then start a discussion during our next developer meeting or feel free to add a new comment to this Pull Request.
-----------------------------------------------------------------------------------------------------------------------------------------
1. Installation

2. Contribution
On how to contribute to devonfw: 
https://github.com/devonfw/.github/blob/master/CONTRIBUTING.asciidoc

First steps regarding Forks or Cloning of Repos and creating branches as well as some git commands:
https://github.com/firstcontributions/first-contributions

3. Create a new commandlet
Where to add the license for a new commandlet:
https://github.com/devonfw/ide/blob/master/documentation/LICENSE.asciidoc

4. Settings
Important notes on configuration and especially configuration files influencing the users local devon-IDE: 
https://github.com/devonfw/ide/blob/master/documentation/configuration.asciidoc

Table of Variables included in the devon-IDE:
https://github.com/devonfw/ide/blob/master/documentation/variables.asciidoc

5. Tipps and tricks

**Comment by cinnamon-coder-hub** [comment]
Hier ist ein kürzerer Text den ich zur Bash-Programmierung zusammengestellt habe. Nächste Woche werde ich wahrscheinlich noch ein paar Links einbauen zu Themen die ich noch als wichtig und hilfreich erachte und die etwas fortgeschrittenere Themen behandeln, wie beispielsweise die verschiedenen Bedeutungen von "$" zusammen mit anderen Zeichen. Dabei will ich allerdings auch nicht den erstbesten Link nehmen (ggf. ändere ich auch unten nochmal was an den Verlinkungen) Die Sprachform wählte ich eher locker und direkt, da ich selbst häufig das Gefühl hatte das solche Tutorials mir den meisten Mehrwert brachten und dem User das Gefühl geben an die Hand genommen und persönlicher eingeführt zu werden. Das ist aber sicher Geschmackssache. Allgemein können wir gerne diskutieren was hier zielführend ist und was definitiv noch rein sollte. Eine knackige Einweisung in Bash in dieser oder anderer Form scheint mir aber definitiv sinnvoll. 

## 5. Tipps and Tricks

### Bash (Linux's Borne Again Shell)
**For beginners:**

You probably know this black box on Windows (the Command Prompt) that accepts commands like 'dir' that shows the content of the directory you are currently in,
or 'cd' with which you can change the directory you are currently in and many more commands that allow you to interact with your Windows Operating System in a more direct and eventually faster way.  
Now for users of a Linux Distribution there is something similar, but way more powerfull, usually called "Bourne Again Shell", or in short BASH. This Command Line Interface or "Shell" is what we use most often when implementing something new for our Command-Line-Tool with the name devon-IDE.  
[Have a look at Bash in 150 Seconds](https://www.youtube.com/watch?v=I4EWvMFj37g) 
For MacOS you often also use BASH. That's why it is often enough to implement code for Mac or Linux (at least to some extend as you'll see soon enough) to be able to use it on both Operating Systems.

Let's get to the practical part of the BASH-usage. 

**If you are using Windows**, *then please install WSL so you will be able to use Bash as well. E.g. [watch this](https://www.youtube.com/watch?v=qYlgUDKKK5A) guide untill Minute 5:58 for a basic setup on Windows and some command examples.*

On the internet are some really good ressources for Bash-beginners.
That's what I now want to make use of by sharing [this robust introduction](https://www.youtube.com/watch?v=oxuRxtrO2Ag), going through some of the most important commands and concepts in round about 70 Minutes (you can find a list of the commands introduced by Joe at the videos description). There are some really usefull commands, that I would have liked to know back then when I started with Bash. When you have learned the knowledge from this video, then you'll be a big step further on your journey with Bash. 
I think it is really worth watching such a video and to directly play around with those commands on your own Shell simultaneously.  This makes the learning process faster and also way more interesting than simply reading or listenning to such a tutorial.  

When you used to program in some other languages, you for sure know how helpful a google search in combination with the results on stackexchange can be. If not, then google your question regarding Bash as usual and mostly you'll find a way to get together the code you want. 
Sometimes it needs more patience to get an acceptable result or to find out that a command you already knew includes what you need, if you add the proper tag to its execution. Take the commands and some of their tags mentioned in [this rather brief tutorial](https://www.freecodecamp.org/news/linux-command-line-bash-tutorial/) as an example. Also definetely try to use the help function from time to time, e.g. type in `ls --help` and you'll see how much more the command `ls` can offer. On the other side, especialy at the beginning, it is better to get an overview of available tags by googling e.g. 
*ls tags bash* to get a way prettier overview plus examples regarding a commands tags like [here](https://www.tecmint.com/15-basic-ls-command-examples-in-linux/) for `ls`.

**If you have some experience with the BASH or directly want to see more:**

Sometimes you just don't know the meaning of a tag or a sign combination (like `$1`, or `$()`) and also don't know how to find its meaning, because google won't directly search for signs. Then a [syntax cheatsheet](https://www.pcwdld.com/bash-cheat-sheet) like this can spare you some searching time.

**Comment by hohwille** [comment]
Great that we have absolute valuable input for this.
However, I have the following cents:
* the PR itself currently only adds an empty documentation so the collected content from the comments should go there
* "Getting started" is the wrong title for this documentation. You have to look at the devonfw-ide from different roles and perspectives. What you want to provide here is a guide for developers who want to contribute to devonfw-ide, whereas developers using devonfw-ide would expect a "Getting started" guide to tell them how to get started in *using* devonfw-ide. As only a very little part of our users (unfortunately) also want to develop and improve devonfw-ide we should IMHO pick a  different title such as "Contributing to devonfw-ide" or "Getting started as developer contributing to devonfw-ide".

**Review by alfeilex — APPROVED** [review]
Bzgl. WSL Installation hat Jörg in Teams letztens darauf hingewiesen, dass wir Rancher-Desktop nutzen sollten, um WSL2 aufzusetzen. 

In Zeile 46 wird ein Video für die Einrichtung verlinkt. Evtl. mit Jörg absprechen, ob man den Guide im Video nutzen kann.

File: documentation/devonfw-ide-contribution-getting-started.asciidoc
Comment (hohwille) [inline documentation/devonfw-ide-contribution-getting-started.asciidoc:68]: I do not get this.
Simply use `git-bash` and you have a full bash for windows without the need for WSL.

File: documentation/devonfw-ide-contribution-getting-started.asciidoc
Comment (hohwille) [inline documentation/devonfw-ide-contribution-getting-started.asciidoc]: Sounds a little bit more like a personal review on the youtube videos (`I think ...`).
IMHO you could simple list a bullet point of links and add once sentence to give the reader an idea what to expect and if it is worth watching for him or not.

**Review by hohwille — APPROVED** [review]
@all thanks for your contribution and work on this valuable piece of documentation.
I did some last small improvements (esp. for the ide-mirrors) as well as proper AsciiDoc-Linking formatting, spelling conventions, etc. - see also our AsciiDoc guidelines as a learning for the future: https://github.com/devonfw/docgen/blob/master/documentation/guidelines.asciidoc

**Comment by hohwille** [comment]
@cinnamon-coder-hub and @CREITZ25 your reviews are requested here by somebody of the team.
This currently blocks the merge.
I would be happy if we can merge this soon and from that point on continue with smaller improvement PRs on the existing documentation.
If you have nothing to add simply approve without comment.

File: documentation/devonfw-ide-contribution-getting-started.asciidoc
Comment (cinnamon-coder-hub) [reply documentation/devonfw-ide-contribution-getting-started.asciidoc]: I made a compromise with a commit a few minutes ago if you are ok with that, by mentioning `git-bash` as first choice, as we primarily use it during our development. As addition I mentioned `WSL` as it can be of value. For example, when using doIsLinux `WSL` will return a 0, whereas `git-bash` won't, what can be usefull while testing. Still, maybe this argument is to weak to keep mentioning WSL?

File: documentation/devonfw-ide-contribution-getting-started.asciidoc
Comment (cinnamon-coder-hub) [reply documentation/devonfw-ide-contribution-getting-started.asciidoc]: The advantage of that would definitely be a fast overview of the resources mentioned. 
On the other side I wanted to accompany the reader a bit more, by not being to short.  Especially for people that are not so familiar with bash-coding yet this would be more comfortable to read and to process. I would also not say, that this particular video is definitely the best around the web (and I'm happy for other sources, that may fit our needs in a better way), but the producer seems to have a good approach for beginners.
Still, if you or the others think that it is to much, then I'll be fine to just make a list of the resources with some brief notes next to them.

**Review by cinnamon-coder-hub — APPROVED** [review]

**Review by CREITZ25 — APPROVED** [review]

**Comment by tobka777** [comment]
I added a section to mirrors about matching order and using variables in mirrors.

File: documentation/devonfw-ide-contribution-getting-started.asciidoc
Comment (hohwille) [inline documentation/devonfw-ide-contribution-getting-started.asciidoc:111]: These hints should IMHO go here instead,:
https://github.com/devonfw/ide/blob/master/documentation/advanced-tooling-generic.asciidoc

Leave the section about advanced tooling here and simply put a link to the existing [documentation](https://github.com/devonfw/ide/blob/master/documentation/advanced-tooling-generic.asciidoc).

File: documentation/devonfw-ide-contribution-getting-started.asciidoc
Comment (hohwille) [inline documentation/devonfw-ide-contribution-getting-started.asciidoc:160]: FYI: I never put detailed links on `devonfw.com` from github. These links have never been stable. The link will be broken within weeks, months or at least after one year.
The entire devonfw.com website is continuously changing and another big redesign in already planned.
I read documentation on github where links are stable.

File: documentation/advanced-tooling-generic.asciidoc
Comment (hohwille) [inline documentation/advanced-tooling-generic.asciidoc:26]: This is windows specific and already documented here:
https://github.com/devonfw/ide/blob/master/documentation/advanced-tooling-windows.asciidoc#create-symbolic-links

File: documentation/advanced-tooling-generic.asciidoc
Comment (hohwille) [inline documentation/advanced-tooling-generic.asciidoc:32]: Also Windows specific and should be documented here:
https://github.com/devonfw/ide/blob/master/documentation/advanced-tooling-windows.asciidoc#tabs-for-cmd

BTW: Heise is a German media and we have many readers who do not speak German.

**Review by hohwille — CHANGES_REQUESTED** [review]
Actually I wanted to merge this already as the new guide is helpful.
Meanwhile advanced tooling was also updated but I have some review remarks that need to be reworked before merge.

File: documentation/advanced-tooling-generic.asciidoc
Comment (cinnamon-coder-hub) [reply documentation/advanced-tooling-generic.asciidoc:26]: Fair enough, should I leave a link here or delete it completely? There is a link to this point inside the getting-started document. I would propose to delete it here completely and to move at least the example to the getting-started-document instead, as quite some people (especially those reading the getting-started document) may not really know what those symbolic links are and we may need to explain newcomers again and again how helpful this is especially for the development of commandlets like mentioned in the example above.

Comment (hohwille) [reply documentation/advanced-tooling-generic.asciidoc:26]: > I would propose to delete it here completely and to move at least the example to the getting-started-document instead ...

makes sense to me.

File: documentation/advanced-tooling-generic.asciidoc
Comment (hohwille) [inline documentation/advanced-tooling-generic.asciidoc:25]: This is still missplaced here. Please remove.
Feel free to improve here:
https://github.com/devonfw/ide/blob/master/documentation/advanced-tooling-windows.asciidoc#tabs-for-cmd

**Review by hohwille — APPROVED** [review]
@cinnamon-coder-hub thanks for addressing the review feedback.
Now ready for merge.

File: documentation/advanced-tooling-generic.asciidoc
Comment (hohwille) [reply documentation/advanced-tooling-generic.asciidoc]: Thanks for the improvement. Done 👍

File: documentation/advanced-tooling-generic.asciidoc
Comment (hohwille) [reply documentation/advanced-tooling-generic.asciidoc]: Thanks for the improvement. Done 👍

File: documentation/advanced-tooling-generic.asciidoc
Comment (hohwille) [reply documentation/advanced-tooling-generic.asciidoc]: Thanks for the improvement. Done 👍

**Pull request merged**
